holidays/countries/new_zealand.py

```diff
@@ -45,27 +46,71 @@ class NewZealand(ObservedHolidayBase, ChristianHolidays, InternationalHolidays,
         "WGN",  # Greater Wellington / Te Pane Matua Taiao
         "WKO",  # Waikato
         "WTC",  # West Coast / Te Tai o Poutini
+        # Subregions:
+        # https://www.govt.nz/browse/work/public-holidays-and-work/public-holidays-and-anniversary-dates/
+        "South Canterbury",
```

Comment:
Unofficial code "STC" was previously used. If we have holidays for this province anyway, maybe we could use it? @arkid15r, how strictly do we follow ISO 3166-2?

Reply:
The current version is not that strictly follows the standard. I hope this will change in`v1`. We probably should think of a way to keep it flexible to allow "extras" built on top of official ISO compliant entities.